catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveProxyPlugin.java

```diff
@@ -36,6 +42,27 @@ public Object doAs(
       UserGroupInformation proxyUser =
           UserGroupInformation.createProxyUser(
               PrincipalUtils.getCurrentPrincipal().getName(), currentUser);
+
+      if (UserGroupInformation.isSecurityEnabled()) {
+        if (ops != null) {
+
+          String token =
+              ops.getClientPool()
+                  .run(
+                      client -> {
+                        return client.getDelegationToken(
+                            currentUser.getUserName(),
+                            PrincipalUtils.getCurrentPrincipal().getName());
+                      });
```

Comment:
Did you get delegation token every time you do the action?

Reply:
Yes, we will have many users. It's useless if we cache the token.